src/components/side_panel/side_panel/side_panel_store.ts

```diff
@@ -0,0 +1,36 @@
+import { SpreadsheetStore } from "../../../stores";
+
+interface SidePanelProps {
+  onCloseSidePanel?: () => void;
+  [key: string]: unknown;
+}
+
+export class SidePanelStore extends SpreadsheetStore {
+  isOpen: boolean = false;
+  props: SidePanelProps = {};
+  component: string = "";
+
+  open(panel: string, panelProps: SidePanelProps = {}) {
+    if (this.isOpen && panel !== this.component) {
+      this.props?.onCloseSidePanel?.();
+    }
+    this.component = panel;
+    this.props = panelProps;
```

Comment:
`this.props` make it look like it's a component, not a store 😅 

nitpick: Also `this.component` being a string sounds strange